### PR TITLE
Black autoformatting

### PR DESCRIPTION
**Pull request by Edgar-21**
Two things.

1. `black --line-length 79  .` was run to format all python files with black, and with a line length of 79 characters, per PEP8. Black by default uses 88.
2. An action that checks that code has been formatted such that applying `black --line-length 79  .` does not result in any changes has been added. It runs on pushes, PRs, and manaully.

When/if a `pyproject.toml` is authored it should include black args, and the action should be updated to reference `pyproject.toml`, rather than including the necessary args in the yaml file.

### CODE REVIEW

**Review by gonuke — APPROVED** [review]
Thanks @Edgar-21 - hopefully this helps us avoid too much whitespace churn in reviews.

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+      
+      - name: Install Black
+        run: |
+          python -m pip install --upgrade pip
+          pip install black
+      
+      - name: Run Black check
```
Comment (gonuke) [inline .github/workflows/black.yml:20]: Why not use the action provided by `black`: https://black.readthedocs.io/en/stable/integrations/github_actions.html

**Review by gonuke — CHANGES_REQUESTED** [review]
Just found this action provided by PSF that might make life even easier

File: .github/workflows/black.yml (hunk 1 above)
Comment (Edgar-21) [reply .github/workflows/black.yml]: much nicer, thank you

Comment (Edgar-21) [reply .github/workflows/black.yml]: actually not sure if the change I just made is exactly what you had in mind, this is all relatively mysterious to me still

**Review by gonuke — APPROVED** [review]
Thanks for updating to the action - let's see if this does what we need.

**Pull request merged**